backend/main/src/Services/InvitationHashService/InvitationHashService.ts

```diff
@@ -127,7 +118,49 @@ export class InvitationHashService implements IGenerateAnInvitationHashService {
 			expiryDatetime,
 		});
 	}
+
+	/**
+	 * This method join a user to the group using an invitation hash.
+	 * This method checks the existence of the invitation hash and the user.
+	 * If the invitation hash and the user exists, this method adds the user to the group.
+	 * And return the groupId of the group.
+	 * @param hash - The hash string of the invitation link hash.
+	 * @param userId - The userId of the user who will be joined the group.
+	 */
+	async joinToGroupUsingAnInvitationHash(
+		hash: string,
+		userId: string,
+	): Promise<
+		Result<{ groupId: string }, UserIdError | UserNotFound | HashNotFound>
+	> {
+		if (!isValidUUIDV4(userId)) {
+			return Err(new UserIdError("UserId is not valid."));
+		}
+
+		const [invitationLinkHash, userData] = await Promise.all([
+			this.nishikiDynamoDBClient.getInvitationLink(hash),
+			this.nishikiDynamoDBClient.getUser({ userId }),
+		]);
+
+		if (!userData) {
+			return Err(new UserNotFound("Requesting user is not found."));
+		}
+
+		if (!invitationLinkHash) {
+			return Err(new HashNotFound("Invalid hash is requested."));
+		}
+
```

Comment:
I think we need check if the invitationLinkHash.linkExpiryTime has already passed. and if the link expires, the specific error should be returned.